test/v2/controllers/requestParsers/validators/CreateForeignPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -2467,10 +2115,62 @@ class CreateForeignPropertyPeriodSummaryValidatorSpec extends UnitSpec with Mock
     "return multiple errors" when {
       "request supplied has multiple errors" in {
         val validator = setUpValidator()
-        validator.validate(CreateForeignPropertyPeriodSummaryRawData("A12344A", "20178", "232301", requestBodyJson)) shouldBe
+        val result    = validator.validate(CreateForeignPropertyPeriodSummaryRawData("A12344A", "20178", "232301", requestBodyJson))
+        result shouldBe
           List(NinoFormatError, TaxYearFormatError, BusinessIdFormatError)
       }
     }
   }
 
+  trait TestData {
```

Comment:
This doesn't need to be in its own trait; nothing's really gained, but it does have some drawbacks, mainly:
1. turns the test into an instance of a class, so you have to deal with private/public vals, specifying types etc
2. "requestBody" ends up with multiple instance creations (one per TestData subclass) for no particular reason
3. it's just added complexity
Easier to just put requestBody as a private val.  Better would be to make it a private def, and parameterise it for the fromDate & toDate.